proc_info.sh

```diff
@@ -14,4 +14,78 @@ function usage(){
   echo "-c                   Show command that started the process"
   echo "-f                   Show opened files by the process"
   echo "-h                   Show this help"
+  exit 0
 }
+
+#Function for -u flag
+get_owner(){
+  pid=$1
+  user_id=$(cat /proc/$pid/status | grep Uid | awk '{ print $2 }')
+  #https://unix.stackexchange.com/a/36582/468165 - Get user by Id
+  getent passwd "$user_id" | cut -d: -f1
+}
+
+#Function for -c flag
+get_command(){
+  pid=$1
+  cat /proc/$pid/comm
+}
+
+#Function for -e flag
+get_env_vars(){
+  pid=$1
+  cat /proc/$pid/environ | tr '\0' '\n'
+}
+
+#Function for -f flag
+get_opened_files(){
+  pid=$1
+  ls -l /proc/$pid/fd | less | awk '{ print $11 }'
+}
+
+while test $# -gt 0; do
+  case "$1" in
+    -h) usage ;;
+    -u)
+      shift
+      if test $# -gt 0; then
+        get_owner $1
+      else
+        echo "no process specified"
+        exit 1
+      fi
+      shift
+      ;;
+    -c)
+      shift
+      if test $# -gt 0; then
+        get_command $1
+      else
+        echo "no process specified"
+        exit 1
+      fi
+      shift
+      ;;
+    -e)
+      shift
+      if test $# -gt 0; then
+        get_env_vars $1
+      else
+        echo "no process specified"
+        exit 1
+      fi
+      shift
+      ;;
+    -f)
+      shift
+      if test $# -gt 0; then
+        get_opened_files $1
+      else
+        echo "no process specified"
+        exit 1
+      fi
+      shift
+      ;;
+    *) usage ;;
```

Comment:
usually when you specify parameter that not exists, program should exit with non-zero return codes. 
For `usage` function it's ok to return `0` exit codes, all other cases should be marked as error.
Check out https://www.cyberciti.biz/faq/bash-get-exit-code-of-command/